src/Services/Air/AirParser.js

```diff
@@ -1392,6 +1396,117 @@ function availability(rsp) {
   };
 }
 
+function getKeyObject(obj, blockKey) {
+  const block = obj[blockKey];
+  const [key] = Object.keys(block);
+
+  return block[key];
+}
+
+function formCouponBlock(coupon) {
+  return {
+    uapi_emd_coupon_ref: coupon.Key,
+    number: parseInt(coupon.Number, 10),
+    status: coupon.Status,
+    svcDesc: coupon.SvcDescription,
+    consumedAtIssuanceInd: coupon.ConsumedAtIssuanceInd === 'true',
+    rfiCode: coupon.RFIC,
+    rfiSubcode: coupon.RFISC,
+    rfiDesc: coupon.RFIDescription,
+    origin: coupon.Origin,
+    destination: coupon.Destination,
+    flightNumber: coupon.FlightNumber,
+    isRefundable: coupon.NonRefundableInd !== 'true',
+  };
+}
+
+function formPassengerBlock(passenger) {
+  return {
+    lastName: passenger['air:NameInfo'].Last,
+    firstName: passenger['air:NameInfo'].First,
+    ageCategory: passenger.TravelerType,
+    age: passenger.Age,
+  };
+}
+
+function getEMDListItem(obj) {
+  const passenger = obj['air:EMDTravelerInfo'];
+  const summary = getKeyObject(obj, 'air:EMDSummary');
+  const coupon = getKeyObject(summary, 'air:EMDCoupon');
+
+  return {
+    summary: {
+      coupon: formCouponBlock(coupon),
+      uapi_emd_ref: summary.Key,
+      number: parseInt(summary.Number, 10),
+      isPrimaryDocument: summary.PrimaryDocumentIndicator === 'true',
+      associatedTicket: summary.AssociatedTicketNumber,
+      platingCarrier: summary.PlatingCarrier,
+      issuedAt: summary.IssueDate,
+    },
+    passenger: formPassengerBlock(passenger)
+  };
+}
+
+function getEMDList(obj) {
+  const eMDSummaryInfo = obj['air:EMDSummaryInfo'];
+
+  if (!eMDSummaryInfo) {
+    processUAPIError.call(this, obj, 'Unable to retrieve EMD list');
+  }
+
+  return Object.values(eMDSummaryInfo).map(val => (getEMDListItem.call(this, val)));
+}
+
+function getEMDItem(obj) {
+  const emdInfo = getKeyObject(obj, 'air:EMDInfo');
+
+  const passenger = emdInfo['air:EMDTravelerInfo'];
+  const supplierLocator = emdInfo[`common_${this.uapi_version}:SupplierLocator`] || [];
+  const emd = getKeyObject(emdInfo, 'air:ElectronicMiscDocument');
+  const coupon = getKeyObject(emd, 'air:EMDCoupon');
+  const payment = getKeyObject(emdInfo, `common_${this.uapi_version}:Payment`);
+  const fop = getKeyObject(emdInfo, `common_${this.uapi_version}:FormOfPayment`);
```

Comment:
I think the usage of `getKeyObject` has to do something with parsing rules I mentioned above